repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java

```diff
@@ -2725,48 +2725,54 @@ public void repairAccesscontrolAlias(String guid) throws AtlasBaseException {
-        if (CollectionUtils.isEmpty(vertices)) {
-            return;
-        }
+        try {
+            List<AtlasVertex> vertices = this.entityGraphMapper.linkBusinessPolicy(policyGuid, linkGuids);
+            if (CollectionUtils.isEmpty(vertices)) {
+                return;
+            }
 
-        // Prepare the response for the entity mutations
-        EntityMutationResponse entityMutationResponse = new EntityMutationResponse();
-        for (AtlasVertex vertex : vertices) {
-            // Convert each vertex to an AtlasEntityHeader and add it to the response
-            AtlasEntityHeader entityHeader = entityRetriever.toAtlasEntityHeader(vertex);
-            entityMutationResponse.addEntity(UPDATE, entityHeader);
+            handleBusinessPolicyMutation(vertices);
+        } catch (Exception e) {
+            LOG.error("Error during linkBusinessPolicy for policyGuid: {}", policyGuid, e);
+            throw e;
+        } finally {
+            RequestContext.get().endMetricRecord(metric);
         }
+    }
 
-        // Collect the vertices into a map for easier access by their GUID property
-        Map<String, AtlasVertex> vertexMap = vertices.stream()
-                .collect(Collectors.toMap(vertex -> vertex.getProperty("__guid", String.class), vertex -> vertex));
-
-        // Notify the policy change notifier about the entities that were mutated
-        this.atlasAlternateChangeNotifier.onEntitiesMutation(entityMutationResponse, vertexMap);
+    @Override
+    @GraphTransaction
+    public void unlinkBusinessPolicy(String policyGuid, Set<String> unlinkGuids) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("unlinkBusinessPolicy.GraphTransaction");
+        try {
+            List<AtlasVertex> vertices = this.entityGraphMapper.unlinkBusinessPolicy(policyGuid, unlinkGuids);
+            if (CollectionUtils.isEmpty(vertices)) {
+                return;
+            }
 
-        // End the performance metrics recording
-        RequestContext.get().endMetricRecord(metric);
+            handleBusinessPolicyMutation(vertices);
+        } catch (Exception e) {
+            LOG.error("Error during unlinkBusinessPolicy for policyGuid: {}", policyGuid, e);
+            throw e;
+        } finally {
+            RequestContext.get().endMetricRecord(metric);
+        }
     }
 
+    private void handleBusinessPolicyMutation(List<AtlasVertex> vertices) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("handleBusinessPolicyMutation");
+        vertices.forEach(vertex -> {
+            Map<String, Object> attributes = new HashMap<>(0);
+            attributes.put(ASSET_POLICY_GUIDS, vertex.getMultiValuedSetProperty(ASSET_POLICY_GUIDS, String.class));
+            attributes.put(ASSET_POLICIES_COUNT, vertex.getPropertyValues(ASSET_POLICIES_COUNT, Long.class));
+        });
```

Comment:
I think this block is redundant

Reply:
If this is true, remove this method & directly call the following wherever required 
`this.atlasAlternateChangeNotifier.onEntitiesMutation(vertices);`